shared/podman/images.go

```
@@ -6,48 +6,170 @@ package podman
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
+	"io"
+	"os"
+	"path"
 	"strings"
 
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
+	"github.com/uyuni-project/uyuni-tools/shared/types"
 	"github.com/uyuni-project/uyuni-tools/shared/utils"
 )
 
+var rpmImageDir = "/usr/share/suse-docker-images/native/"
+
 // Ensure the container image is pulled or pull it if the pull policy allows it.
-func PrepareImage(image string, pullPolicy string, args ...string) error {
+//
+// Returns the image name to use. Note that it may be changed if the image has been loaded from a local RPM package.
+func PrepareImage(image string, pullPolicy string, args ...string) (string, error) {
 	log.Info().Msgf("Ensure image %s is available", image)
 
-	needsPull, err := checkImage(image, pullPolicy)
+	if strings.ToLower(pullPolicy) == "always" {
+		log.Debug().Msgf("Pulling image %s because pull policy is 'always'", image)
+		return image, pullImage(image, args...)
+	}
+
+	present, err := IsImagePresent(image)
 	if err != nil {
-		return err
+		return image, err
+	}
+
+	if present {
+		log.Debug().Msgf("Image %s already present", image)
+		return image, nil
+	}
+
+	rpmImageFile, _ := GetRpmImagePath(image)
```

I don't like the idea to totally ignore an error: this can hide real errors

---

removed it